### ISSUE TITLE
LegATo

### ISSUE THREAD

**Issue by aubreyodom** [Bioconductor/Contributions#3524]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/wejlab/LegATo

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by lshep** [comment]
ggeffects does appear in your DESCRIPTION and is available on CRAN;  we will investigate why it did not install properly on our system.

**Comment by DarioS** [comment]
The package has a good variety of functions and documentation. Some issues have been identified.

- Installation instructions should use BiocManager, which works with GitHub. Use the text below.

````
```
BiocManager::install("wejlab/LegATo")
```
````

- Variable and function naming uses snake case e.g. `create_formatted_MAE`, `counts_dat` but [camelCase is expected](https://contributions.bioconductor.org/r-code.html?#variable-names).

- Shouldn't `tempfolder <- tempfile()` be `tempfolder <- tempdir()`?

- The package introduces a new data set. Justify why this is necessary instead of using existing ones, such as provided by the Bioconductor package [microbiomeDataSets](https://bioconductor.org/packages/release/data/experiment/html/microbiomeDataSets.html). Reuse is desirable.

- Many recent packages that deal with microbiome data allow [TreeSummarizedExperiment](https://bioconductor.org/packages/3.19/bioc/html/TreeSummarizedExperiment.html) container class. Please support it.

- For each bar chart or heatmap, add one or two sentences of interpretation after it.

- Packages aren't allowed to have pre-generated results (e.g. Dolosigranulum_timepoint.png).

- There is only one unit test, which is a spell check. Unit tests should test for algorithmic correctness.

```r
spelling::spell_check_test(vignettes = TRUE, error = FALSE, skip_on_cran = TRUE)
```

- Something is wrong with the plots. For example,

![image](https://github.com/user-attachments/assets/dfca5b8d-3a28-4fc4-8224-685dbcb867cf)

- Use `vapply` or `mapply` instead of `for`.

```r
  new.hsv <- c()
  for (i in seq_len(num.vs)) {
    temp <- rbind(hues.hsv[1, ], s[i], v[i])
    new.hsv <- cbind(new.hsv, temp)
  }
```

could be more efficiently be written using `mapply`.

**Comment by aubreyodom** [comment]
@DarioS Please see our replies below.

> 1. Installation instructions should use BiocManager, which works with GitHub. Use the text below.
BiocManager::install("wejlab/LegATo")

Thank you for the suggestion. We have updated the packaged README.md file to use the Bioconductor install function.

> 2. Variable and function naming uses snake case e.g. create_formatted_MAE, counts_dat but camelCase is expected.

We appreciate the suggestion. We adhere to the the Hadley Wickham advice to use "snake case" with underscores to separate words within a name and only use lowercase letters:

https://style.tidyverse.org/syntax.html?q=snake%20case#object-names

One study has found that subjects recognize identifiers in the underscore style over camel case more quickly (https://doi.org/10.1109/ICPC.2010.41).

> 3. Shouldn't tempfolder <- tempfile() be tempfolder <- tempdir()?

We have changed this line of code in the vignette to use the tempdir() function.

> 4. The package introduces a new data set. Justify why this is necessary instead of using existing ones, such as provided by the Bioconductor package microbiomeDataSets. Reuse is desirable.

The dataset in this project is indeed an existing dataset, whose originating publication motivated the creation of this package. The included data is small and fast to use, making it an efficient choice for computing examples. The total combined size of the data included in the package is a mere 197.3 kb, or 1/5 of a megabyte. Therefore, I do not believe it necessary to download data from a Bioconductor package given the ease of use with the currently included dataset.

> 5. Many recent packages that deal with microbiome data allow TreeSummarizedExperiment container class. Please support it.

We have added functionality to the `create_formatted_MAE` function for users to create a MultiAssayExperiment from a TreeSummarizedExperiment container class.

> 6. For each bar chart or heatmap, add one or two sentences of interpretation after it.

These have been added to the vignette.

> 7. Packages aren't allowed to have pre-generated results (e.g. Dolosigranulum_timepoint.png).

We have changed the Generalized Estimating Equations section of the vignette to print graphics from the output files saved in the temporary folder location. 

> 8. There is only one unit test, which is a spell check. Unit tests should test for algorithmic correctness.

Unit tests have been added that cover 83.03% of the package.

> 9. Something is wrong with the plots.

The plots in the vignette (the ones in question) have been fixed.

> 10. Use vapply or mapply instead of for.
  new.hsv <- c()
  for (i in seq_len(num.vs)) {
    temp <- rbind(hues.hsv[1, ], s[i], v[i])
    new.hsv <- cbind(new.hsv, temp)
  }
could be more efficiently be written using mapply.

We have rewritten this to completely omit an apply step, with the following code:
new.hsv <- dplyr::tibble(V1 = rep(hues.hsv[1, ], num.vs),
                           V2 = rep(s, each = length(hues)),
                           V3 = rep(v, each = length(hues))) |>
      t() |> as.matrix() |> magrittr::set_rownames(NULL)

**Comment by lshep** [comment]
The default branch of your GitHub repository has been added to Bioconductor's
git repository as branch devel.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/aubreyodom.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("LegATo")`. The package 'landing page' will be created at

https://bioconductor.org/packages/LegATo

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
